docs/design/pattern_matching.md

```diff
@@ -53,7 +60,7 @@ not permitted as proper patterns, so cannot contain bindings.
 
```

Comment:
Can we drop some text from / rewrite the overview? Once we have an overloading design it might make sense to add some of that back, but for now it seems out of place to talk about overload resolution here.

Reply:
How's this?